src/main/kotlin/com/bonitasoft/gradle/maven/settings/MavenSettingsPlugin.kt

```diff
@@ -128,24 +129,24 @@ class MavenSettingsPlugin : Plugin<Project> {
     }
 
     private fun registerMirrors(project: Project) {
+        val mirrorsToAdd = mutableMapOf<String, Action<MavenArtifactRepository>>()
         project.repositories.toList().filter { repo ->
             if (repo.name.equals(ArtifactRepositoryContainer.DEFAULT_MAVEN_LOCAL_REPO_NAME)) {
                 return@filter false
             }
             repo.getMirror(settings)?.also { mirror ->
-                if (!project.repositories.names.contains(mirror.id)) {
-                    project.repositories.maven { repo ->
-                        repo.name = mirror.id
-                        repo.url = URI.create(mirror.url)
-                        project.logger.info("Replaced '${repo.name}' with mirror ${mirror.id} configured in maven's settings.xml")
-                    }
-                } else {
-                    project.logger.info("Replaced '${repo.name}' with mirror ${mirror.id} (already created) configured in maven's settings.xml")
+                project.logger.info("Replaced '${repo.name}' with mirror '${mirror.id}' configured in maven's settings.xml")
+                mirrorsToAdd.putIfAbsent(mirror.id) { repo ->
+                    repo.name = mirror.id
+                    repo.url = URI.create(mirror.url)
                 }
             } != null
         }.forEach {
             project.repositories.remove(it)
```

Comment:
Why do we remove all repositories?

Reply:
We remove all the repositories that have a mirror matching ( the previous function call is a filter ) 
That's not very visible

Reply:
Ok, got it.